File: hard/roman-numeral-conversion/original.py
import sys

s = sys.argv[1]

# write your solution here
translations = {
  "I": 1,
  "V": 5,
  "X": 10,
  "L": 50,
  "C": 100,
  "D": 500,
  "M": 1000
}
number = 0
s = s.replace("IV", "IIII").replace("IX", "VIIII")
s = s.replace("XL", "XXXX").replace("XC", "LXXXX")
s = s.replace("CD", "CCCC").replace("CM", "DCCCC")
for char in s:
    number += translations[char]
print(number)

File: hard/roman-numeral-conversion/original.js
const s = process.argv[2];

// write your solution here
let values = { 'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000 };
let sum = values[s.charAt(s.length - 1)];
for (let i = s.length - 2; i >= 0; i--) {
  let first = s.charAt(i); 
  let second = s.charAt(i + 1);
  if (values[first] < values[second]) {
    sum -= values[first];
  } else {
    sum += values[first];
  }
}

console.log(sum);